### DISCUSSION TITLE
Make native window more configurable

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#640] [Ideas / Feature Requests]
@Skjolbir came up with this feature request in #576. There a many more parameters which could be passed in 

https://github.com/zauberzeug/nicegui/blob/9093c72331b55f988157a57ffe64ff464e2f640e/nicegui/native_mode.py#L17-L19

The first idea was to provide a `webview_args` parameter to `ui.run`. But that would further blow up the already long list of parameters. Also the `webview.create_window` has some parameters which would be helpful good to manipulate from the users.

Maybe it is better to introduce an `app.native` object which has properties like `create_window_args`, `webview_start_args` and an callback `on_webview_started` which provides the window instance from `create_window`. This could be used to add menus etc.

**Comment by rodja** [comment]
Ideally this would also simplify implementation of #634 which adds window-icon support for Windows.

**Comment by rodja** [comment]
Maybe 

```py
ui.native.args_to_create_window = { 'resizable': False }
```

**Reply by Skjolbir** [reply]
Would that be set prior to ui.run ?
Seems a good way to keep ui.run args clean/concise.

**Reply by falkoschindler** [reply]
Current draft:

```py
app.native.start_args = { ... }
app.native.window_args = { ... }
```

**Comment by rodja** [comment]
This would also solve #665.

**Comment by smojef** [comment]
As an inspiration from a developer's perspective, I like the [magic underscore notation](https://plotly.com/python/creating-and-updating-figures/#magic-underscore-notation) from plotly API ;-)
It offers the best of both worlds:
- clean nested structures, and
- simple top level configuration of nested values.

**Reply by falkoschindler** [reply]
Interesting idea! Our uvicorn parameters in `ui.run` are already grouped in a similar fashion.
